### ISSUE TITLE
App crashes when during OAuth2 request using swedish electronic identification

### ISSUE THREAD

**Issue by lkmill** [a85/POSTMan-Chrome-Extension#5965]
Please read through the [guidelines](https://github.com/postmanlabs/postman-app-support#guidelines-for-reporting-issues) before creating a new issue.

**Describe the bug**
We are working against an API that requires an access token received from an OAuth2 login. The OAuth2 end point authenticates users using the Swedish electronic identification app BankID. There are two types of BankID authentication apps/methods. One is called "Mobile BankID" where a user identifies by supplying a social security number that BankID which then sends a request to the persons Mobile BankID app in their phone where the user supplies there login code. This works fine. The other method involves an app installed on the computer the user is using. To use this the website has a button "Open BankID app" which is supposed to initiate the app on the computer. Upon clicking this button Postman dies.

This occurs in Linux, MacOS and Windows.

**To Reproduce**

It is not possible to reproduce this behaviour without having access to this specific API, and i cannot supply the details required to use it. I have however started postman through a terminal so i can see it's output. Upon termination the following error stacks are printed:

```
1550853830992 main info ["OAuth2WindowManager\\x7estartLoginWith - Opening auth login window","URL REMOVED"]
1550853831237 main info ["OAuth2WindowManager\\x7estartLoginWith - Received redirect on auth login window"]
1550853831238 main info ["OAuth2WindowManager\\x7estartLoginWith - URL did not match the registered callbackURL, so skipping"]
1550853831373 main info ["OAuth2WindowManager\\x7estartLoginWith - Received redirect on auth login window"]
1550853831377 main info ["OAuth2WindowManager\\x7estartLoginWith - URL did not match the registered callbackURL, so skipping"]
1550853831710 main info ["OAuth2WindowManager\\x7estartLoginWith - Received redirect on auth login window"]
1550853831712 main info ["OAuth2WindowManager\\x7estartLoginWith - URL did not match the registered callbackURL, so skipping"]
1550853832837 main info ["OAuth2WindowManager\\x7estartLoginWith - Received redirect on auth login window"]
1550853832838 main info ["OAuth2WindowManager\\x7estartLoginWith - URL did not match the registered callbackURL, so skipping"]
1550853832977 main info ["OAuth2WindowManager\\x7estartLoginWith - Received redirect on auth login window"]
1550853832978 main info ["OAuth2WindowManager\\x7estartLoginWith - URL did not match the registered callbackURL, so skipping"]
1550853835096 main info ["OAuth2WindowManager\\x7estartLoginWith - Received redirect on auth login window"]
TypeError: Cannot read property 'join' of undefined
    at handleUrlChange (/opt/postman/app/resources/app/services/OAuth2WindowManager.js:73:27)
    at WebContents.<anonymous> (/opt/postman/app/resources/app/services/OAuth2WindowManager.js:126:7)
    at emitTwo (events.js:130:20)
    at WebContents.emit (events.js:213:7)
Main~handleUncaughtError - Uncaught errors TypeError: Cannot read property 'join' of undefined
    at handleUrlChange (/opt/postman/app/resources/app/services/OAuth2WindowManager.js:73:27)
    at WebContents.<anonymous> (/opt/postman/app/resources/app/services/OAuth2WindowManager.js:126:7)
    at emitTwo (events.js:130:20)
    at WebContents.emit (events.js:213:7)
1550853835114 main error ["Main\\x7ehandleUncaughtError - Uncaught errors",{"name":"TypeError","message":"Cannot read property 'join' of undefined","stack":"TypeError: Cannot read property 'join' of undefined\n    at handleUrlChange (/opt/postman/app/resources/app/services/OAuth2WindowManager.js:73:27)\n    at WebContents.<anonymous> (/opt/postman/app/resources/app/services/OAuth2WindowManager.js:126:7)\n    at emitTwo (events.js:130:20)\n    at WebContents.emit (events.js:213:7)"}]
Error: Object has been destroyed
    at Object.stateChangeHandler (/opt/postman/app/resources/app/services/windowManager.js:731:33)
    at invokeFunc (/opt/postman/app/resources/app/node_modules/lodash/lodash.js:10333:23)
    at trailingEdge (/opt/postman/app/resources/app/node_modules/lodash/lodash.js:10382:18)
    at Timeout.timerExpired [as _onTimeout] (/opt/postman/app/resources/app/node_modules/lodash/lodash.js:10370:18)
    at ontimeout (timers.js:469:11)
    at tryOnTimeout (timers.js:304:5)
    at Timer.listOnTimeout (timers.js:264:5)
Main~handleUncaughtError - Uncaught errors Error: Object has been destroyed
    at Object.stateChangeHandler (/opt/postman/app/resources/app/services/windowManager.js:731:33)
    at invokeFunc (/opt/postman/app/resources/app/node_modules/lodash/lodash.js:10333:23)
    at trailingEdge (/opt/postman/app/resources/app/node_modules/lodash/lodash.js:10382:18)
    at Timeout.timerExpired [as _onTimeout] (/opt/postman/app/resources/app/node_modules/lodash/lodash.js:10370:18)
    at ontimeout (timers.js:469:11)
    at tryOnTimeout (timers.js:304:5)
    at Timer.listOnTimeout (timers.js:264:5)
1550853835785 main error ["Main\\x7ehandleUncaughtError - Uncaught errors",{"name":"Error","message":"Object has been destroyed","stack":"Error: Object has been destroyed\n    at Object.stateChangeHandler (/opt/postman/app/resources/app/services/windowManager.js:731:33)\n    at invokeFunc (/opt/postman/app/resources/app/node_modules/lodash/lodash.js:10333:23)\n    at trailingEdge (/opt/postman/app/resources/app/node_modules/lodash/lodash.js:10382:18)\n    at Timeout.timerExpired [as _onTimeout] (/opt/postman/app/resources/app/node_modules/lodash/lodash.js:10370:18)\n    at ontimeout (timers.js:469:11)\n    at tryOnTimeout (timers.js:304:5)\n    at Timer.listOnTimeout (timers.js:264:5)"}]
```

**Expected behavior**
Expected the app not to terminate.

**App information (please complete the following information):**
 - App Type Native App
 - Postman Version [e.g. 6.7.3]
 - OS: Linux, MacOS and Windows 10

**Comment by harryi3t** [comment]
@lohfu Is there a way we can try out this authentication with this App? If yes, could you provide us the link to the documentation? Also, we would appriciate any other information which would help us to reproduce this.

**Comment by numaanashraf** [comment]
Fixed in Postman v7.0.8
